### PR TITLE
add shift handover/reminder notifications

### PR DESCRIPTION
**Pull request by nadade**
things to do still:

- remove slackr (and only use the native sdk)
- add tests

wider todo areas:
- refactor config setup
- get rid of callbacks

### CODE REVIEW

File: src/pagerduty.js [hunk 1]
```diff
@@ -275,11 +278,54 @@ function processSchedulesFromConfig(done) {
   });
 }
 
+function getEngineersOncall(scheduleId, escalationPolicyId, done) {
```
Comment (abtris) [inline src/pagerduty.js]: I think that can have better name what about `getOncallForHandover` ?

File: src/pagerduty.js [hunk 2]
```diff
@@ -275,11 +278,54 @@ function processSchedulesFromConfig(done) {
...
+  const extraOptions = {
+    qs: {
+      'include[]': 'users',
+      'escalation_policy_ids[]': escalationPolicyId,
+    },
+  };
+  getSchedule(scheduleId, extraOptions, (err, scheduleInfo) => {
+    if (err) { return done(err); }
+    const timeToCheck = moment(nconf.get('TIME_SINCE')) || moment();
```
Comment (abtris) [inline src/pagerduty.js]: this is that time of handover?

File: src/slack-api.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    .then((res) => {
+      if (res.ok === false) { return done(new Error(`Slack returned an error: ${res.error}`)); }
+      debug('Message sent: ', res.ts);
+      return done();
+    }).catch((err) => {
+      done(err);
+    });
+}
+
+function getUserList(done) {
```
Comment (abtris) [inline src/slack-api.js]: This is for mapping Slack user at PD users?

Comment (nadade) [reply src/slack-api.js]: @abtris yes, this is to map slack IDs and user emails (to connect PD and Slack users by email)

File: src/pagerduty.js (hunk 2 above)
Comment (nadade) [reply src/pagerduty.js]: this is the time of execution of the task; I'm using the legacy function that returns oncalls for the whole configured interval (2 weeks), can be optimized in the next iteration

File: src/pagerduty.js (hunk 1 above)
Comment (nadade) [reply src/pagerduty.js]: how about `getEngineersOncallForHandover`? as this returns the users (plus their relevant oncalls)

File: bin/pdoverrides [hunk 4]
```diff
@@ -38,15 +42,56 @@ function runCheck(configFile) {
   });
 }
 
+function sendReminder(cmd){
```
[the remark this answers is not in the document]
Comment (nadade) [reply bin/pdoverrides]: @abtris do you mean there should be two commands, `sendReminder` and `sendHandover`?

[the remark this answers is not in the document]
Comment (abtris) [reply bin/pdoverrides]: no, I can just make easy run as file `bin/pdoverrides` and `bin/pdhandover` but that can be in this way too

**Pull request merged**